### ISSUE TITLE
Segfault when analyzing code through entry point with k-induction

### ISSUE THREAD

**Issue by pmatos** [esbmc/esbmc#248]
K-Induction seems to a cause a segfault in the following program (reduced):
(`numarith.c`)
```
void a();
b() {
  for (;;)
    a();
}
void a() { b(); }
```

```
$ esbmc --k-induction --function b numarith.i
ESBMC version 6.4.0 64-bit x86_64 linux
file numarith.i: Parsing
numarith.i:2:1: warning: type specifier missing, defaults to 'int' [-Wimplicit-int]
b() {
^
Converting
Generating GOTO Program
GOTO program creation time: 0.084s
[1]    107607 segmentation fault (core dumped)  esbmc --k-induction --function b numarith.i
```

Using esbmc 6.4.0.

**Comment by pmatos** [comment]
Thanks for the quick fix. I will give it a go again later and get back to you.

**Comment by pmatos** [comment]
This seems to be fixed but the same example file generates a double free corruption. Opening separate error.

**Comment by pmatos** [comment]
It's not completely clear why esbmc ends with:
```
Runtime decision procedure: 0.000s
The forward condition is unable to prove the property
Unable to prove or falsify the program, giving up.
VERIFICATION UNKNOWN
``` 
on the above example though.

**Comment by lucasccordeiro** [comment]
ESBMC was unable to falsify (via the `base case`) or prove (via the `forward condition` or `inductive step`) the target property.

You could use the option `--unlimited-k-steps` to check whether ESBMC can falsify or prove this target property.

Further information about the `base case`, `forward condition`, and `inductive step` can be found at:

https://ssvlab.github.io/lucasccordeiro/papers/sttt2017.pdf

**Comment by mikhailramalho** [comment]
> It's not completely clear why esbmc ends with:
> 
> ```
> Runtime decision procedure: 0.000s
> The forward condition is unable to prove the property
> Unable to prove or falsify the program, giving up.
> VERIFICATION UNKNOWN
> ```
> 
> on the above example though.

We are not actually running inductive step here, only the base case and forward condition. 

Currently, our k-induction does not support recursion, so the inductive step is disabled. Even if you use `--unlimited-k-steps` esbmc won't be able to verify the program, because the forward condition step can't fully unroll the infinite loop/recursion in the program.

**Comment by mikhailramalho** [comment]
Fixed by PR #249
